src/include/btree_inline.h

```diff
@@ -1963,6 +1965,16 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_splitp)
         return (false);
     }
 
+    /* Don't evict the disaggregated page that should belong to the next checkpoint. */
+    if (modified && F_ISSET(btree, WT_BTREE_DISAGGREGATED) &&
+      btree->checkpoint_gen == __wt_gen(session, WT_GEN_CHECKPOINT))
+        /*
+         * TODO: we only know that the btree has been visited by the latest checkpoint. But we don't
```

Comment:
Does this mean that we cannot evict a page from a table in between checkpoints? I'm wondering if there is something that we could do based on `conn->disaggregated_storage.global_checkpoint_id`, from which you know whether the next checkpoint has been opened.

Reply:
Yes, currently we cannot evict. I was thinking about that yesterday and I think we need to remember the last completed checkpoint id.

Reply:
I think I have a better idea. Updated the code.